Merge docs-uplift feature branch into master

jwilson-mdb opened the pull request:
## DESCRIPTION
Merges `docs-uplift` into `master`

## STAGING
https://deploy-preview-246--docs-relational-migrator.netlify.app/

Notable/Large changes:

- [Overview](https://deploy-preview-246--docs-relational-migrator.netlify.app/getting-started/) revised to act as product overview. Previously a general _process_ overview, which will soon be covered in the upcoming Quickstart doc.
- [Connect to Databases](https://deploy-preview-246--docs-relational-migrator.netlify.app/database-connections/)
  - updates these docs to be database specific
  - now includes [using migrator without a live db connection](https://deploy-preview-246--docs-relational-migrator.netlify.app/database-connections/migrator-without-live-connection/) to improve discoverability
- [Installation](https://deploy-preview-246--docs-relational-migrator.netlify.app/installation/) section revised to:
  - Have the landing page act as a concept page, detailing the types of installations and considerations.
  - Flatten documentation for production installation. 3 related tasks -> 1 complete task
  - Grouping all required content together. File locations for each OS are given on the installation pages.
- Remaining ToC entries revised to group at the task level (Data Modeling, Data Migration ...) rather than concept level (Mapping rules, Diagrams ...)

## JIRA
https://jira.mongodb.org/browse/DOCSP-48961

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

davidsoulmdb commented:
I notice that when I click on the Connect to Databases LHS menu, there is no longer a page load just the menu expansion. has this page been intentionally hidden/moved? I can't find it so wanted to confim this is intentional

jwilson-mdb commented:
> I notice that when I click on the Connect to Databases LHS menu, there is no longer a page load just the menu expansion. has this page been intentionally hidden/moved? I can't find it so wanted to confim this is intentional

Good catch @davidsoulmdb, thanks! This was unintentional. I pushed a fix and the Connect to Databases page should now appear when you click on the menu.

sarah-olson-mongodb commented on source/installation.txt:
```diff
@@ -10,68 +10,141 @@ Installation
+
+`Apache Kafka <https://kafka.apache.org/>`__ is an open-source platform for distributed 
+workloads. If you use {+rel-mig+} for critical production workloads or long-running 
+:abbr:`CDC (Change Data Capture)` syncs, MongoDB recommends Kafka. Using {+rel-mig+}
```
(nit)
"MongoDB recommends Kafka integration." Or "using Kafka". Or similar.

jwilson-mdb replied:
Updated to `using Kafka`

sarah-olson-mongodb commented on source/installation.txt:
```diff
@@ -10,68 +10,141 @@ Installation
 
-Relational Migrator supports local deployment and unattended server deployment models.
-For instructions on installing Relational Migrator, see the following pages:
+.. _rm-deployment-considerations:
```
(parallelism)
Suggestion.
On this page we describe the Local Desktop, Unattened Server/VM, Kafka Cluster, and Kafka on Confluent Cloud installations. In the TOC we have L2 headings for for Development and Production installations. Can you find some ways to tie these together to create a little more parallelism between the two?

jwilson-mdb replied:
This one is admittedly a bit blurred. David and I went through several revisions of this - the short version is that, although local install is intended mostly for development, it is appropriate for migrating small production workloads.

I'm going to publish this as is for now, and make a note to follow up with David to see if we can revise the wording to parallel the L2s more. I don't want to make edits without looping in David/RM team as this seems to be an important doc for user onboarding, and had the most input from the engineering team re: specific language.

sarah-olson-mongodb commented on source/installation/kafka-deployments/migrator-with-kafka.txt:
```diff
@@ -1,8 +1,8 @@
-Integrating Relational Migrator with Kafka
-==========================================
+======================
+Kafka Deployment Guide
```
(parallelism)
Maybe just "Kafka Deployments". We don't use "Guide" anywhere else in the Migrator library.

jwilson-mdb replied:
Updated

sarah-olson-mongodb commented on source/installation/kafka-deployments/migrator-with-kafka.txt:
```diff
@@ -16,41 +16,35 @@ Integrating Relational Migrator with Kafka
-MongoDB cluster that acts as a data sink.
-
-Architecturally, Relational Migrator is deployed within Kafka as a Kafka 
+Architecturally, {+rel-mig+} is deployed within Kafka as a Kafka 
```
(consistency)
Below, in an unedited part. You might consider: ... change data capture (CDC). This will keep consistent with other pages introducing CDC.

sarah-olson-mongodb commented on source/getting-started.txt:
```diff
@@ -18,69 +19,175 @@ Get Started with Relational Migrator
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+If your application must remain online to support reads and writes
+during the migration process, you can use Change Data Capture (CDC) to
```
(terminology)
A quick google search suggests that change data capture isn't used as a proper noun, usually. :)

sarah-olson-mongodb commented on source/installation/kafka-deployments/migrator-with-kafka.txt:
```diff
@@ -1,8 +1,8 @@
-Integrating Relational Migrator with Kafka
-==========================================
+======================
+Kafka Deployment Guide
```
(question) 
Does it make sense to move the Kafka deployment information to an L2 heading?

jwilson-mdb replied:
For now, I'm leaning towards leaving it as an L3 because it is a type of production installation. I think the idea here is to steer new users and if we have "Kafka" on the same level as "Development" and "Production", it is not immediately clear where Kafka fits in.

This might be a good point to revisit when we take a second look at the language in the "Deployment Models" on the installation page. I'll add them both to my notes to follow up on.

sarah-olson-mongodb commented on source/installation.txt:
```diff
@@ -10,68 +10,141 @@ Installation
+.. _rm-deployment-considerations:
+
+Supported Deployment Models
+---------------------------
```
(question) 
What are the differences you're specifying between Installation and Deployment? Are you using them interchangeably or is there a specific difference?

jwilson-mdb replied:
To me, this is very closely related to the other comments on parallelism and moving the Kafka docs to an L2.
I'll group these comments into some notes and either ticket it out or schedule time with David to go over.

sarah-olson-mongodb commented on source/installation/application-settings.txt:
```diff
@@ -0,0 +1,52 @@
+Relational Migrator and select :guilabel:`Settings`:
+
+.. figure:: /img/projects/settings-panel-location.png 
+  :alt: Settings panel location for Relational Migrator
```
(image)
Can we make the image a little bigger? This is pretty hard to see. You could potentially do that just by minimizing the dark space on the left?

There is also functionality that allows you to click to enlarge, which might be helpful here!

jwilson-mdb replied:
I cropped this image to eliminate the empty space

jwilson-mdb commented on source/getting-started.txt:
```diff
@@ -18,69 +19,175 @@ Get Started with Relational Migrator
+For more information, see :ref:`<rm-use-cases>`.
+
+{+rel-mig+} integrates with Apache Kafka and Confluent Cloud for large scale 
+migrations. For more information, see :ref:`<kafka-intro>`.
```
I will skip this one as I'm changing the title of the linked page from `Kafka Deployment Guide` to `Kafka Deployments`

jwilson-mdb commented on source/getting-started.txt:
```diff
@@ -18,69 +19,175 @@ Get Started with Relational Migrator
+Use Cases
+---------
+
+{+rel-mig+} supports both Snapshot and Continuous migration jobs.
```
Updated `migration jobs` plaintext to be a link that points to the `Types of Migration Jobs` [section](https://deploy-preview-246--docs-relational-migrator.netlify.app/jobs/sync-jobs/#types-of-migration-jobs)

davidsoulmdb replied:
Optional suggestion: mention that snapshot jobs can also be used for batched jobs on getting-started

Snapshot: Snapshot migration jobs run once, typically for a point in time migration strategy. ->
Snapshot: Snapshot migration jobs run once and shut down. Suitable for migrating a full database at a point in time, batch migrations, and incremental data syncs eg nightly runs. Larger database migrations may be split into batches using id or date-based table filters and run in parallel for improved performance.

jwilson-mdb replied:
I tweaked the wording a bit, but updated the snapshot description to read:
```
Snapshot migration jobs migrate all data once, and then stop. You can 
use snapshot migrations for point-in-time database migrations,
batch migrations, and incremental syncs. You can also use them with 
:ref:`table filters <rm-table-filters>` to split large migration jobs 
into parallel batches to improve performance. 
```

jwilson-mdb commented on source/getting-started.txt:
```diff
@@ -18,69 +19,175 @@ Get Started with Relational Migrator
+connection speed.
+
+After the migration completes, take your legacy application offline and
+launch your new application.
```
As far as I know, we don't have any content covering this. The upcoming "Migration Planning Guide" doc would be the likely target. I'll make a note in my planning doc to crosslink these scenario sections with corresponding guide sections.

jwilson-mdb commented on source/getting-started.txt:
```diff
@@ -18,69 +19,175 @@ Get Started with Relational Migrator
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+If your application must remain online to support reads and writes
+during the migration process, you can use Change Data Capture (CDC) to
```
Updated!

jwilson-mdb commented on source/getting-started.txt:
```diff
@@ -18,69 +19,175 @@ Get Started with Relational Migrator
+cause :abbr:`CDC (Change Data Capture)` to run indefinitely without
+completing, which is not recommended with {+rel-mig+}.
+
+{+rel-mig+} will support more migration scenarios and use cases
```
Removed

jwilson-mdb commented on source/getting-started.txt:
```diff
@@ -18,69 +19,175 @@ Get Started with Relational Migrator
+
+{+rel-mig+} supports the following target databases:
+
+.. include:: /includes/table-supported-mongodb-versions.rst
```
Added one in the source DB's section, and a direct link to MongoDB connection doc in the target dbs section.

jwilson-mdb commented on source/database-connections.txt:
```diff
@@ -12,35 +12,56 @@ Database Connections
+  your relational database.
+
+- Create a project using a sample relational database schema.
+
```
Added

jwilson-mdb commented on source/database-connections/mysql.txt:
```diff
@@ -0,0 +1,95 @@
+
+.. _rm-mysql-connection-string:
+
+Database Connection String
```
https://jira.mongodb.org/browse/DOCSP-49257

jwilson-mdb commented on source/database-connections/mysql.txt:
```diff
@@ -0,0 +1,95 @@
+
+.. procedure::
+   :style:  normal
+
```
https://jira.mongodb.org/browse/DOCSP-49258

jwilson-mdb commented on source/database-connections/mysql.txt:
```diff
@@ -0,0 +1,95 @@
+
+   .. include:: /includes/database-connections/connect-to-db-steps-part-1.rst
+
+   .. step:: To create a connection string, enter the following:
```
I am struggling to get the parser to ingest my changes correctly here due to the use of multiple includes within the procedure directive. As this is non-blocking, I've created https://jira.mongodb.org/browse/DOCSP-49261 to follow up.

jwilson-mdb commented on source/database-connections/mysql.txt:
```diff
@@ -0,0 +1,95 @@
+
+- {+rel-mig+} relies on the open-source Debezium connector to capture 
+  row-level changes. For more details, see
+  `Debezium MySQL <https://debezium.io/documentation/reference/stable/connectors/mysql.html>`__.
```
The period shows up in my preview, but it is before the icon indicating it is an external link

jwilson-mdb commented on source/database-connections/mysql.txt:
```diff
@@ -0,0 +1,95 @@
+      
+      .. include:: /includes/table-mysql-connection-fields.rst
+
+   .. include:: /includes/database-connections/connect-to-db-steps-part-2.rst
```
Updated

jwilson-mdb commented on source/projects/create-project-loading-schema-files.txt:
```diff
@@ -116,12 +116,12 @@ to generate DDL files from your relational system.
 Steps
 -----
 
```
Will be covered in https://jira.mongodb.org/browse/DOCSP-49258

jwilson-mdb commented on source/projects/create-project-sample-schema.txt:
```diff
@@ -38,7 +38,7 @@ Before you can run migration jobs from the sample schema you must either:
+   schema presented in {+rel-mig+}.
 
 2. Download and execute the 
    `Northwind.sql <https://github.com/pthom/northwind_psql/blob/master/northwind.sql>`__ 
```
Moved, updated ToC, and added redirects from old URL to new

jwilson-mdb commented on source/database-connections/migrator-without-live-connection.txt:
```diff
@@ -0,0 +1,53 @@
+
+You can create a project without a live relational database connection 
+to visualize and model your data or to evaluate Relational Migrator. However,
+Relational Migrator requires a live database connection to migrate data.
```
For this and the related comment on the getting started page, I'd like to do this as a follow up. I want to be careful with the language as a big reason for moving this doc in the ToC and highlighting it as an alternative is that:

- Many customers did not know you could use RM without a connection
- Lots of customers do not have access to their live DB

I want to chat with David and make sure we aren't too prescriptive with use case language. I'd hate to have a non-exhaustive list that could imply its not appropriate for some case we missed.

davidsoulmdb commented:
Overall could publish as-is, I do haave some minor wording tweaks to suggest but could also hold off until v2?

installation - LGTM

getting-started/ optional suggstions?

1. Databases supported with a [JDBC driver](https://deploy-preview-246--docs-relational-migrator.netlify.app/getting-started/#std-label-jdbc-databases) should name them please? 
YugabyteDB, 
CockroachDB, 
SAP Sybase IQ (SAP IQ), 
SAP SQL Anywhere

2. Separate comment in https://github.com/mongodb/docs-relational-migrator/pull/246#discussion_r2038600425

For discussion on database-connections/: 
1. Is it more skimmable if we lead with the database name in database-connections/ ?
To connect to a source relational database::

Connect to MySQL -> MySQL Database Guide
Connect to DB2-> DB2 Database Guide

2. We should also at least mention that the JDBC databases by name? Note that Artem is preparing connection strings as a separate DOCSP. eg. You can also use our JDBC connectivity to connect to YugabyteDB, CockroachDB, SAP Sybase IQ (SAP IQ), and SAP SQL Anywhere.

3. Clarify that MDB can only be a destination:

To connect to the destination database, see [Connect to MongoDB.](https://deploy-preview-246--docs-relational-migrator.netlify.app/database-connections/mongodb/#std-label-rm-connect-to-mongo)

Without live db page:
1. Clarify what Northwind is? You can try Relational Migrator without a relational database by using our demonstration schema. Northwind is a popular example database for an order management system.

davidsoulmdb reviewed (approved):
LGTM but have suggested final minor wording edits as a separate comment - optional

jwilson-mdb commented:
> Overall could publish as-is, I do haave some minor wording tweaks to suggest but could also hold off until v2?
> 
> installation - LGTM
> 
> getting-started/ optional suggstions?
> 
> 1. Databases supported with a [JDBC driver](https://deploy-preview-246--docs-relational-migrator.netlify.app/getting-started/#std-label-jdbc-databases) should name them please?
>    YugabyteDB,
>    CockroachDB,
>    SAP Sybase IQ (SAP IQ),
>    SAP SQL Anywhere

The linked section has all supported JDBC dbs listed in the table.

> 2. Separate comment in [Merge docs-uplift feature branch into master #246 (comment)](https://github.com/mongodb/docs-relational-migrator/pull/246#discussion_r2038600425)
> 
Updated!

> For discussion on database-connections/:
> 
> 1. Is it more skimmable if we lead with the database name in database-connections/ ?
>    To connect to a source relational database::
> 
> Connect to MySQL -> MySQL Database Guide Connect to DB2-> DB2 Database Guide
> 

If you are referring to the titles on those procedural pages, our [docs style guide](https://www.mongodb.com/docs/meta/style-guide/information-types/prototypes/task/#task-page-prototype) specifies that task page titles should begin with a present tense action verb.
As far as the link text itself, our [style guide](https://www.mongodb.com/docs/meta/style-guide/style/links-and-cross-references/#make-link-text-clear) also prefers that link text correspond to the destination page title.

> 2. We should also at least mention that the JDBC databases by name? Note that Artem is preparing connection strings as a separate DOCSP. eg. You can also use our JDBC connectivity to connect to YugabyteDB, CockroachDB, SAP Sybase IQ (SAP IQ), and SAP SQL Anywhere.

I think this change makes more sense to include with the JDBC connection strings DOCSP ticket. Adding it to the `Getting Started` section of this doc may be confusing for users when we don't have docs to help them get started with that task yet.

> 3. Clarify that MDB can only be a destination:
> 
> To connect to the destination database, see [Connect to MongoDB.](https://deploy-preview-246--docs-relational-migrator.netlify.app/database-connections/mongodb/#std-label-rm-connect-to-mongo)
> 

Updated to use your suggestion

> Without live db page:
> 
> 1. Clarify what Northwind is? You can try Relational Migrator without a relational database by using our demonstration schema. Northwind is a popular example database for an order management system.

Tweaked the wording a bit, but updated to explain what northwind is 
```
You can try Relational Migrator without a relational database by using the pre-defined sample schema. The sample schema uses the ``Northwind`` database,
a popular example database for an order management system.
```
